components/utils/escapeRegExp.js

```diff
@@ -0,0 +1,10 @@
+/**
+ * Escape special characters for RegEx
+ * @param   {String}    term
+ * @returns {Boolean}
+ */
+const escapeRegExp = term => {
+    return term.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
```

Comment:
Just a minor thing, arrow functions have an implicit return, so the `return` statement can be omitted, instead you could just use:
```
const escapeRexExp = term => term.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
```

iirc either prettier of eslint should catch this, do you have them installed in your editor? (although maybe we never set those rules)